### PR TITLE
Add fn<->win inverter

### PR DESCRIPTION
**Pull request by TGODiamond**
Adds a new field in `msi_ec_fn_win_swap_conf`, a bool named `invert`. Set to true if the fn<->win config files report the opposite direction.

### CODE REVIEW

**Comment by TGODiamond** [comment]
PR is ready to be merged! :)

**Comment by TGODiamond** [comment]
There we go! Ready, for real this time!

**Comment by teackot** [comment]
Hi! Thanks for the PR

I'm thinking we can make the code more readable by merging the `ec_set_bit()` and `ec_unset_bit()` into a single `ec_set_bit(u8 addr, u8 bit, bool value)` function, which just choses between the set_bit and unset_bit macros. This way we can avoid both the nested ifs (in your first version) and very long lines with ternaries (they don't like long lines in the kernel). It'll also simplify the `[mic]mute_led_sysfs_set()` functions

**Comment by glpnk** [comment]
We can make EC_RS_BIT(addr, bit, val) and EC_RS_BITS(addr, bit(?), size(?), val), which should write single bit and several bits

**Comment by teackot** [comment]
Doesn't seem like we win anything by doing this. The performance impact of different bit manipulation algos is negligible as the functions are called very rarely. So I think we should prioritize readability and keep bit manipulations simple

**Comment by glpnk** [comment]
Other kernel modules might share own bit manipulation algo, so we can use it

**Comment by teackot** [comment]
I looked for examples similar to mine. None of them use dedicated (un)set_bit macros, but they use an `if`

[one](https://github.com/torvalds/linux/blob/83a7eefedc9b56fe7bfeff13b6c7356688ffa670/drivers/platform/x86/intel/pmc/core.c#L578), and basically the same: [two](https://github.com/torvalds/linux/blob/83a7eefedc9b56fe7bfeff13b6c7356688ffa670/drivers/platform/x86/intel/pmt/crashlog.c#L104)

```c
if (reset)
	fd &= ~CNP_PMC_LATCH_SLPS0_EVENTS;
else
	fd |= CNP_PMC_LATCH_SLPS0_EVENTS;
```

A little different, but also manipulates EC: [three](https://github.com/torvalds/linux/blob/83a7eefedc9b56fe7bfeff13b6c7356688ffa670/drivers/platform/x86/eeepc-laptop.c#L1004)

```c
static void eeepc_set_fan_ctrl(int manual)
{
	u8 value = 0;

	ec_read(EEEPC_EC_FAN_CTRL, &value);
	if (manual == EEEPC_FAN_CTRL_MANUAL)
		value |= EEEPC_EC_FAN_CTRL_BIT;
	else
		value &= ~EEEPC_EC_FAN_CTRL_BIT;
	ec_write(EEEPC_EC_FAN_CTRL, value);
}
```

**Comment by TGODiamond** [comment]
> I'm thinking we can make the code more readable by merging the ec_set_bit() and ec_unset_bit() into a single ec_set_bit(u8 addr, u8 bit, bool value) function

Did just that, looks much better now!

**Comment by teackot** [comment]
Great! Just some Kernel-specific style remarks and we can merge it

File: msi-ec.c [hunk 1]
```diff
@@ -1780,7 +1786,11 @@ static int ec_unset_bit(u8 addr, u8 bit)
 	if (result < 0)
 		return result;
 
-	unset_bit(stored, bit);
+	if (value) {
+		set_bit(stored, bit);
+	} else {
+		unset_bit(stored, bit);
+	}
```
Comment (teackot) [inline msi-ec.c:1793]:
```suggestion
	if (value)
		set_bit(stored, bit);
	else
		unset_bit(stored, bit);
```

No braces required here

File: msi-ec.c [hunk 2]
```diff
@@ -2021,9 +2031,13 @@ static ssize_t fn_key_store(struct device *dev, struct device_attribute *attr,
 	int result;
 
 	if (streq(buf, "right")) {
-		result = ec_set_bit(conf.fn_win_swap.address, conf.fn_win_swap.bit);
+		result = ec_set_bit(conf.fn_win_swap.address,
+					conf.fn_win_swap.bit,
+					true ^ conf.fn_win_swap.invert);
```
Comment (teackot) [inline msi-ec.c:2036]:
```suggestion
		result = ec_set_bit(conf.fn_win_swap.address,
				    conf.fn_win_swap.bit,
				    true ^ conf.fn_win_swap.invert);
```

Alignment

File: msi-ec.c [hunk 3]
```diff
@@ -2021,9 +2031,13 @@ static ssize_t fn_key_store(struct device *dev, struct device_attribute *attr,
...
 	} else if (streq(buf, "left")) {
-		result = ec_unset_bit(conf.fn_win_swap.address, conf.fn_win_swap.bit);
+		result = ec_set_bit(conf.fn_win_swap.address,
+					conf.fn_win_swap.bit,
+					false ^ conf.fn_win_swap.invert);
```
Comment (teackot) [inline msi-ec.c:2040]:
```suggestion
		result = ec_set_bit(conf.fn_win_swap.address,
				    conf.fn_win_swap.bit,
				    false ^ conf.fn_win_swap.invert);
```

Alignment

File: msi-ec.c [hunk 4]
```diff
@@ -2053,9 +2067,13 @@ static ssize_t win_key_store(struct device *dev, struct device_attribute *attr,
 	int result;
 
 	if (streq(buf, "right")) {
-		result = ec_unset_bit(conf.fn_win_swap.address, conf.fn_win_swap.bit);
+		result = ec_set_bit(conf.fn_win_swap.address,
+					conf.fn_win_swap.bit,
+					false ^ conf.fn_win_swap.invert);
```
Comment (teackot) [inline msi-ec.c:2072]:
```suggestion
		result = ec_set_bit(conf.fn_win_swap.address,
				    conf.fn_win_swap.bit,
				    false ^ conf.fn_win_swap.invert);
```

More alignment

File: msi-ec.c [hunk 5]
```diff
@@ -2053,9 +2067,13 @@ static ssize_t win_key_store(struct device *dev, struct device_attribute *attr,
...
 	} else if (streq(buf, "left")) {
-		result = ec_set_bit(conf.fn_win_swap.address, conf.fn_win_swap.bit);
+		result = ec_set_bit(conf.fn_win_swap.address,
+					conf.fn_win_swap.bit, 
+					true ^ conf.fn_win_swap.invert);
```
Comment (teackot) [inline msi-ec.c:2076]:
```suggestion
		result = ec_set_bit(conf.fn_win_swap.address,
				    conf.fn_win_swap.bit, 
				    true ^ conf.fn_win_swap.invert);
```

Ditto

File: msi-ec.c [hunk 6]
```diff
@@ -2132,11 +2150,13 @@ static ssize_t cooler_boost_store(struct device *dev,
 
 	if (streq(buf, "on"))
 		result = ec_set_bit(conf.cooler_boost.address,
-				    conf.cooler_boost.bit);
+				    conf.cooler_boost.bit,
+					true);
```
Comment (teackot) [inline msi-ec.c:2154]:
```suggestion
				    conf.cooler_boost.bit,
				    true);
```

File: msi-ec.c [hunk 7]
```diff
@@ -2132,11 +2150,13 @@ static ssize_t cooler_boost_store(struct device *dev,
...
 
 	else if (streq(buf, "off"))
-		result = ec_unset_bit(conf.cooler_boost.address,
-				      conf.cooler_boost.bit);
+		result = ec_set_bit(conf.cooler_boost.address,
+				    conf.cooler_boost.bit,
+					false);
```
Comment (teackot) [inline msi-ec.c:2159]:
```suggestion
		result = ec_set_bit(conf.cooler_boost.address,
				    conf.cooler_boost.bit,
				    false);
```

**Review by teackot — CHANGES_REQUESTED** [review]

**Comment by TGODiamond** [comment]
There you go!

**Comment by TGODiamond** [comment]
Is it really correct formatting? We are mixing tabs and spaces now.

**Comment by teackot** [comment]
Yes, tabs + spaces are allowed to be used for alignment, only tabs can be used for indentation

Thanks! Merging now

**Pull request merged**